modules/Bio/EnsEMBL/Hive/Accumulator.pm

```diff
@@ -131,6 +133,60 @@ sub dataflow {
     }
 }
 
+sub checkEmptyKeys {
+  my ( $self, $key_signature, $accu_address ) = @_;
+  # we get number of brackets, that are empty for key signature and address
+  # and verify that each empty brackets in key_signature is empty in adress
+  my $curvedBrackets = $self->findEmptyBrackets($key_signature, '{', '}');
+  my $curvedBracketsAddress = $self->findEmptyBrackets($accu_address, '{', '}');
+  my %addressBrackets = map { $_ => 1 } @$curvedBracketsAddress;
+
+  foreach my $position ( @$curvedBrackets ) {
+    if (!exists($addressBrackets{$position})) {
+        die "Null hash key in accumulator, empty curved brackets number " . $position;
+    }
+  }
+
+  my $squareBrackets = $self->findEmptyBrackets($key_signature, '[', ']');
+  my $squareBracketsAddress = $self->findEmptyBrackets($accu_address, '[', ']');
+  %addressBrackets = map { $_ => 1 } @$squareBracketsAddress;
+
+  foreach my $position ( @$squareBrackets ) {
+    if (!exists($addressBrackets{$position})) {
+        die "Null array key in accumulator, empty square brackets number " . $position;
+    }
+  }
+}
+
+sub findEmptyBrackets {
```

Comment:
May I suggest something like this:
`sub findEmptyBrackets {
    my ( $self, $string, $open, $close ) = @_;
    my $count = 0;
    my $result = [];

    # look for opening bracket
    while ($string =~ /$open/g) {
        # count how many opening brackets we have
        $count++;
        if ($string =~ /\G(?=$close)/) {
            # store number of bracket if we find an empty pair (like {})
            push (@$result, $count);
        }
    }
    return $result;
}
`

I find the current version a little hard to read.
If you don't like the lookahead assertion (the ?= in the regex), this should be equivalent:
`if (substr($string, pos($string), 1) eq '}');`